polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1089,18 +1094,27 @@ fn filter_backed_statements_from_disabled_validators<T: shared::Config + schedul
 	let mut filtered = false;
 
 	let minimum_backing_votes = configuration::Pallet::<T>::config().minimum_backing_votes;
+	let core_index_enabled = configuration::Pallet::<T>::config()
+		.node_features
+		.get(FeatureIndex::InjectCoreIndex as usize)
+		.map(|b| *b)
+		.unwrap_or(false);
 
 	// Process all backed candidates. `validator_indices` in `BackedCandidates` are indices within
 	// the validator group assigned to the parachain. To obtain this group we need:
 	// 1. Core index assigned to the parachain which has produced the candidate
 	// 2. The relay chain block number of the candidate
 	backed_candidates.retain_mut(|bc| {
-		// Get `core_idx` assigned to the `para_id` of the candidate
-		let core_idx = match scheduled.get(&bc.descriptor().para_id) {
-			Some(core_idx) => *core_idx,
-			None => {
-				log::debug!(target: LOG_TARGET, "Can't get core idx of a backed candidate for para id {:?}. Dropping the candidate.", bc.descriptor().para_id);
-				return false
+		let core_idx = if let Some(core_idx) = bc.assumed_core_index(core_index_enabled) {
+			core_idx
+		} else {
+			// Get `core_idx` assigned to the `para_id` of the candidate
```

Comment:
I think we should check here if we got multiple candidates on different cores for this para here.
If we did and the `InjectCoreIndex` is not enabled, we shouldn't back anything for this para here, because we can't tell if disabled validators voted or not (because we can't know the assigned group).

Reply:
or just filter the votes from all disabled validators of all groups assigned to this para (invalid votes belonging to another group should not be recorded anyway, but we can't check that). with the current code, we'd just randomly use one of them

Reply:
Before this function is called, the `elastic_scaling_mvp_filter` will filter out any candidates that are missing core_index when the `ElasticScalingCoreIndex` feature is enabled.  If the feature is not enabled we assume that the para:core mapping is unique.

I don't think the logic here needs to be altered if a para has multiple cores assigned.

Reply:
oh, right. If `elastic_scaling_mvp_filter` will do this in the future then it's fine